lib/Sema/ConstraintSystem.cpp

```diff
@@ -535,7 +535,12 @@ static void checkNestedTypeConstraints(ConstraintSystem &cs, Type type,
 
 Type ConstraintSystem::openUnboundGenericType(
     Type type, ConstraintLocatorBuilder locator) {
-  assert(!type->hasTypeParameter());
+  if (type->hasTypeParameter()) {
```

Comment:
This is a hack, we need to fix the underlying problem.

Reply:
fixed - see below